### PR TITLE
OcAppleKernelLib: Optionally set kext bundle version

### PR DESCRIPTION
**Pull request by PMheart**
Closes https://github.com/acidanthera/bugtracker/issues/2048

### CODE REVIEW

File: Include/Acidanthera/Library/OcAppleKernelLib.h [hunk 1]
```diff
@@ -878,7 +879,8 @@ PrelinkedInjectKext (
   IN     UINT32             InfoPlistSize,
   IN     CONST CHAR8        *ExecutablePath OPTIONAL,
   IN OUT CONST UINT8        *Executable OPTIONAL,
-  IN     UINT32             ExecutableSize OPTIONAL
+  IN     UINT32             ExecutableSize OPTIONAL,
+  OUT CONST CHAR8           **BundleVersion OPTIONAL
```
[the remark this answers is not in the document]
Comment (PMheart) [reply Include/Acidanthera/Library/OcAppleKernelLib.h:883]: It was automatically fixed by uncrustify.

File: Library/OcAppleKernelLib/CachelessContext.c [hunk 2]
```diff
@@ -945,6 +953,16 @@ CachelessContextAddKext (
         if (!Context->Is32Bit) {
           InfoPlistLibraries = InfoPlistLibraries64;
         }
+      } else if ((BundleVersion != NULL) && (AsciiStrCmp (TmpKeyValue, INFO_BUNDLE_VERSION_KEY) == 0)) {
```
[the remark this answers is not in the document]
Comment (PMheart) [reply Library/OcAppleKernelLib/CachelessContext.c:956]: https://github.com/acidanthera/OpenCorePkg/pull/358/commits/25ea9c80cccf6f8ac1152ae2d247ad0a76347fa2

File: Include/Acidanthera/Library/OcAppleKernelLib.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (PMheart) [reply Include/Acidanthera/Library/OcAppleKernelLib.h]: https://github.com/acidanthera/OpenCorePkg/pull/358/commits/25ea9c80cccf6f8ac1152ae2d247ad0a76347fa2

File: Library/OcAppleKernelLib/CachelessContext.c [hunk 3]
```diff
@@ -945,6 +953,19 @@ CachelessContextAddKext (
...
```
Comment (vit9696) [inline Library/OcAppleKernelLib/CachelessContext.c:956]: You do comparison as part of non-debug code, this is undesired.

File: Library/OcAppleKernelLib/MkextContext.c [hunk 4]
```diff
@@ -1264,8 +1273,30 @@ MkextInjectKext (
...
+      }
+
+      //
+      // Match CFBundleVersion.
+      //
+      if (AsciiStrCmp (TmpKeyValue, INFO_BUNDLE_VERSION_KEY) == 0) {
+        if (PlistNodeCast (KextPlistValue, PLIST_NODE_TYPE_STRING) == NULL) {
+          break;
+        }
+
```
Comment (vit9696) [inline Library/OcAppleKernelLib/MkextContext.c:1292]: Is it safe to return the value here? Worried about use after free.

File: Library/OcAppleKernelLib/CachelessContext.c (hunk 3 above)
Comment (PMheart) [reply Library/OcAppleKernelLib/CachelessContext.c]: https://github.com/acidanthera/OpenCorePkg/commit/d107500d4566e6ac328b94e12ce843074c3a5a22

**Pull request merged**